src/test/java/org/alfresco/heartbeat/HeartBeatTest.java

```diff
@@ -0,0 +1,223 @@
+    {
+        // Disable heartbeat in data collector service ( as if set in prop file)
+        when(mockDataCollectorService.isEnabledByDefault()).thenReturn(false);
+
+        HeartBeat heartbeat = new HeartBeat(context,true);
+
+        // heartbeat enabled in licence
+        LicenseDescriptor mockLicenseDescriptor =  mock(LicenseDescriptor.class);
+        when(mockLicenseDescriptor.isHeartBeatDisabled()).thenReturn(false);
+
+        assertFalse(heartbeat.isEnabled());
+        assertFalse("Job should not be scheduled at this point.",isJobScheduled());
+
+        heartbeat.onLicenseChange(mockLicenseDescriptor);
+
+        // Check heartbeat is disabled and job unscheduled
+        assertTrue(heartbeat.isEnabled());
+        assertTrue("Job should be scheduled.",isJobScheduled());
+    }
+
+    @Test
+    public void testOnLicenceFailRevertsToEnabled()
+    {
+        // Enable heartbeat in data collector service ( as if set in prop file)
+        when(mockDataCollectorService.isEnabledByDefault()).thenReturn(true);
+
+        HeartBeat heartbeat = new HeartBeat(context,true);
+
+        // heartbeat disabled in licence
+        LicenseDescriptor mockLicenseDescriptor =  mock(LicenseDescriptor.class);
+        when(mockLicenseDescriptor.isHeartBeatDisabled()).thenReturn(true);
+        heartbeat.onLicenseChange(mockLicenseDescriptor);
+
+        assertFalse(heartbeat.isEnabled());
+        assertFalse("Job should not be scheduled at this point.",isJobScheduled());
+
+        // Revert back to default state
+        heartbeat.onLicenseFail();
+
+        // Check heartbeat is disabled and job unscheduled
+        assertTrue(heartbeat.isEnabled());
+        assertTrue("Job should be unscheduled.",isJobScheduled());
+    }
+
+    @Test
+    public void testOnLicenceFailRevertsToDisabled()
+    {
+        // Disable heartbeat in data collector service ( as if set in prop file)
+        when(mockDataCollectorService.isEnabledByDefault()).thenReturn(false);
+
+        HeartBeat heartbeat = new HeartBeat(context,true);
+
+        // heartbeat enabled in licence
+        LicenseDescriptor mockLicenseDescriptor =  mock(LicenseDescriptor.class);
+        when(mockLicenseDescriptor.isHeartBeatDisabled()).thenReturn(false);
+        heartbeat.onLicenseChange(mockLicenseDescriptor);
+
+        assertTrue(heartbeat.isEnabled());
+        assertTrue("Job should be scheduled at this point.",isJobScheduled());
+
+        // Revert back to default state
+        heartbeat.onLicenseFail();
+
+        // Check heartbeat is disabled and job unscheduled
+        assertFalse(heartbeat.isEnabled());
+        assertFalse("Job should be unscheduled.",isJobScheduled());
+    }
+
+    private boolean isJobScheduled()
+    {
+        Scheduler scheduler = (Scheduler) context.getBean("schedulerFactory");
+        String[] jobs = {};
+        try
+        {
+            jobs = scheduler.getJobNames( Scheduler.DEFAULT_GROUP);
+        } catch (SchedulerException e)
```

Comment:
the catch statement has to be one line down. But if you take my other device that is not important anymore.